### DISCUSSION TITLE
Model file added to the channel JSONs?

### DISCUSSION THREAD

**Discussion by fiveseven808** [Kilvoctu/aiyabot#156] [Ideas]
I have a lot of models, and the bot is deployed in a handful of servers and channels. 

I don't want every channel and server to have access to every single model I have for various reasons.

I found I can access unlisted models (i.e. models not listed in models.csv, but exists in the webui models folder) by referencing their shorthash in the data_model prompt. I'm not listing this as a bug because I actually like this for my purposes. 

However, if there were a way to assign accessible models to individual channels/servers that would be great!
While we're at it, would it be possible to put the name of the server and channel in the JSON file? that way I don't have to go to the individual server+channel to find out which exact JSON it's using. 

Again, while we're at it, can we also limit external networks per that model.csv style format? i.e. loras, tis, etc.

**Comment by Kilvoctu** [comment]
> if there were a way to assign accessible models  to individual channels/servers

There is not, as far as I know. While slash commands themselves can be limited to certain channels/servers, the choices are registered globally in Discord. There may be some very obscure workaround for it, but I'm not aware of any.  

The /settings command allows bot hosts to set their channel defaults. From my observation, the average user doesn't mess with most of the parameters. If you *really* want to restrict your models usage from end user, the best thing to do in my opinion is to fork aiyabot. Then on your fork's code, remove `data_model` from the /draw command. You'll still be able to set defaults with /settings, but user will not be able to change the model (unless they know about the shorthash thing)

> would it be possible to put the name of the server and channel in the JSON file?

Yeah this is something that inconveniences me, as well. JSON doesn't support comments, but I could simply add a dummy parameter that's never used, where the value is the server+channel.

> can we also limit external networks per that model.csv style format? i.e. loras, tis, etc.  

I know many hosts want to obfuscate what they're sharing, but this won't happen at least until I finish [GUI project](https://github.com/users/Kilvoctu/projects/1). The reason is, typing out entries in models.csv is already a tedious and janky process I'm trying to eliminate. It's also been a cause of many bug reports/Issues for the repo. I don't want to extend this problem to the other things until a better foundation is in place.
